pandas/core/dtypes/cast.py

```diff
@@ -2062,8 +2062,17 @@ def maybe_cast_to_integer_array(
     # doesn't handle `uint64` correctly.
     arr = np.asarray(arr)
 
-    if is_unsigned_integer_dtype(dtype) and (arr < 0).any():
-        raise OverflowError("Trying to coerce negative values to unsigned integers")
+    if is_unsigned_integer_dtype(dtype):
+        try:
+            if (arr < 0).any():
+                raise OverflowError(
+                    "Trying to coerce negative values to unsigned integers"
+                )
+        except TypeError:
+            if (casted < 0).any():
+                raise OverflowError(
```

Comment:
where is this hit in tests?

Reply:
```result = Series(["0", "1", "2"], dtype=uint8)```, this is hit. For all uint in ```any_int_dtype```

Reply:
i dont understand this.  if i pass `["0", "1", "2"]` to np.array on L2046 with dtype=np.uint8 i get back a uint8 ndarray that shouldn't raise here